### PR TITLE
Updated README instructions for installing mpas-bundle on derecho

### PR DESCRIPTION
**Pull request by amstokely**
* __TYPE__: enhancement 

* __KEYWORDS__: README, installation instructions, formatting improvements, automation

* __DESCRIPTION OF CHANGES__:

    - This PR introduces significant enhancements to the README file to improve the installation instructions for the mpas-bundle, particularly focusing on clarity and ease of use to streamline the setup process. The primary changes include:

    - Reformatted Installation Instructions: The installation instructions have been reorganized sensible subsections, where all terminal commands are in code boxes. This makes it easy to copy and paste the commands directly from github, which will installation errors due to typos. 

   

    - Updated Git and Git LFS Recommendations: The section on git usage has been updated to emphasize the importance of creating a .gitconfig file and ensuring Git LFS is installed and enabled before building the mpas-bundle. This update is crucial for streamlining access to multiple repositories and handling large files efficiently.

    - Added logic to the environment setup scripts that deactivates conda if it's activated. This ensures the python packages loaded as modules are use in the build vs. their conda counterparts.

    - Streamlined Build and Test Process: The instructions for building and testing the bundle on compute nodes have been simplified and updated to reflect the use of the configure script and the new batch job submission process.

    - These changes aim to make the installation process more intuitive and error-proof, reducing the barrier to entry for new users and contributing to a smoother setup experience.

* __ISSUE__:  NA

* __LIST OF MODIFIED FILES__:
    - README.md
    - env-setup/gnu-derecho.sh/csh (new)
    -  env-setup/intel-derecho.sh/csh (new)
    - CMakeLists.txt (modified)
    - env-setup/*cheyenne* (deleted)

__TESTS CONDUCTED__: All mpas-jedi unit tests passed for the gnu and intel platforms. The build type tested was RelWithDebInfo. The names of the tests are listed below.

---
```bash
  Test  #1: mpasjedi_coding_norms
  Test  #2: test_mpasjedi_geometry
  Test  #3: test_mpasjedi_state
  Test  #4: test_mpasjedi_model
  Test  #5: test_mpasjedi_increment
  Test  #6: test_mpasjedi_errorcovariance
  Test  #7: test_mpasjedi_linvarcha
  Test  #8: test_mpasjedi_unsinterp_4pe
  Test  #9: test_mpasjedi_geometry_iterator_2d_2pe
  Test #10: test_mpasjedi_geometry_iterator_3d_2pe
  Test #11: test_mpasjedi_getvalues
  Test #12: test_mpasjedi_obslocalization
  Test #13: test_mpasjedi_obslocalization_vertical
  Test #14: test_mpasjedi_obslocalizations
  Test #15: test_mpasjedi_forecast
  Test #16: test_mpasjedi_hofx3d
  Test #17: test_mpasjedi_hofx3d_nbam
  Test #18: test_mpasjedi_hofx3d_ropp
  Test #19: test_mpasjedi_hofx4d
  Test #20: test_mpasjedi_hofx4d_pseudo
  Test #21: test_mpasjedi_convertstate_bumpinterp
  Test #22: test_mpasjedi_convertstate_unsinterp
  Test #23: test_mpasjedi_parameters_bumpcov
  Test #24: test_mpasjedi_parameters_bumploc
  Test #25: test_mpasjedi_dirac_bumpcov
  Test #26: test_mpasjedi_dirac_bumploc
  Test #27: test_mpasjedi_dirac_noloc
  Test #28: test_mpasjedi_3dvar
  Test #29: test_mpasjedi_3dvar_bumpcov
  Test #30: test_mpasjedi_3dvar_bumpcov_nbam
  Test #31: test_mpasjedi_3dvar_bumpcov_ropp
  Test #32: test_mpasjedi_3denvar_bumploc
  Test #33: test_mpasjedi_3denvar_dual_resolution
  Test #34: test_mpasjedi_3denvar_2stream_bumploc
  Test #35: test_mpasjedi_3denvar_amsua_allsky
  Test #36: test_mpasjedi_3denvar_amsua_bc
  Test #37: test_mpasjedi_3dhybrid_bumpcov_bumploc
  Test #38: test_mpasjedi_3dfgat
  Test #39: test_mpasjedi_3dfgat_pseudo
  Test #40: test_mpasjedi_4denvar_ID
  Test #41: test_mpasjedi_4denvar_VarBC
  Test #42: test_mpasjedi_4denvar_VarBC_nonpar
  Test #43: test_mpasjedi_4denvar_bumploc
  Test #44: test_mpasjedi_4dfgat
  Test #45: test_mpasjedi_eda_3dhybrid
  Test #46: test_mpasjedi_rtpp
  Test #47: test_mpasjedi_letkf_3dloc_4pe
  Test #48: test_mpasjedi_lgetkf_4pe
  Test #49: test_mpasjedi_forecast_2pe
  Test #50: test_mpasjedi_parameters_bumpcov_2pe
  Test #51: test_mpasjedi_parameters_bumploc_2pe
  Test #52: test_mpasjedi_3dvar_2pe
  Test #53: test_mpasjedi_3dhybrid_bumpcov_bumploc_2pe

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -39,6 +39,7 @@ ecbuild_bundle( PROJECT rttov     GIT "https://github.com/JCSDA/rttov.git"
 
 ecbuild_bundle( PROJECT oops      GIT "https://github.com/JCSDA/oops.git"        BRANCH develop UPDATE )
 ecbuild_bundle( PROJECT saber     GIT "https://github.com/JCSDA/saber.git"       BRANCH develop UPDATE)
+ecbuild_bundle( PROJECT crtm      GIT "https://github.com/JCSDA/CRTMv3.git"               BRANCH develop UPDATE )
```
Comment (liujake) [inline CMakeLists.txt]: you may make this line better aligned with other lines.

**Comment by liujake** [comment]
For the list of modified files in the PR message, better just use the output from 'git diff --name-status develop'.

**Comment by liujake** [comment]
should you also mention in the README about 'module unload conda' stuff?

File: CMakeLists.txt (hunk 1 above)
Comment (amstokely) [reply CMakeLists.txt]: Agree!

**Comment by liujake** [comment]
Better also mention that the gnu compiler is the recommended compiler for now.

**Comment by amstokely** [comment]
> should you also mention in the README about 'module unload conda' stuff?

Ya good catch. I just added a bullet about that to the pr description.

**Comment by amstokely** [comment]
> Better also mention that the gnu compiler is the recommended compiler for now.

Addressed this in the latest comment.

**Review by liujake — APPROVED** [review]
@amstokely Let me know if you do not have permission to merge. Then I can merge.

**Comment by amstokely** [comment]
@liujake I don't have permission to merge, can you ? Thanks!

**Review by liujake — APPROVED** [review]

**Pull request merged**